src/charm.py

```diff
@@ -378,6 +378,15 @@ def _reconcile(self):
             insecure_skip_verify=insecure_skip_verify,
         )
 
+        # Add custom processors from Juju config
+        if processors_raw := cast(str, self.config.get("processors")):
+            for processor_name, processor_config in yaml.safe_load(processors_raw).items():
+                self.otel_config.add_processor(
+                    name=processor_name,
+                    processor_config=processor_config,
+                    pipelines=["metrics", "logs", "traces"],
+                )
```

Comment:
Our `_reconcile` method is becoming a monster. We need to refactor it as soon as possible.

Before addressing #57, can't we extract this feature-specific lines to a separate method?